packages/smol/components/sections/Revoke/useAllowances.tsx

```diff
@@ -0,0 +1,584 @@
+	 ** obtain allowances for each address in the list. This process allows us to gather allowance
+	 ** data for all tokens listed.
+	 *********************************************************************************************/
+	const tokenAddresses = useMemo(() => {
+		return isTokensLoading ? [] : listTokensWithBalance(currentChainSyncEntry?.chainID).map(item => item.address);
+	}, [currentChainSyncEntry?.chainID, isTokensLoading, listTokensWithBalance]);
+
+	/**********************************************************************************************
+	 ** The allowances vary across different chains, necessitating us to reset the current state
+	 ** when the user switches chains or change the address.
+	 *********************************************************************************************/
+	useAsyncTrigger(async () => {
+		if (
+			!cachedChainSync ||
+			cachedChainSync.some(item => item.address === address && item.chainID === safeChainID) ||
+			!address
+		) {
+			return;
+		}
+
+		addChainSyncEntry({address, chainID: isDev ? chainID : safeChainID, blockNumber: 0n});
+		set_entryNonce(nonce => nonce + 1);
+		set_chainSyncNonce(nonce => nonce + 1);
+	}, [cachedChainSync, address, addChainSyncEntry, chainID, safeChainID]);
+
+	/**********************************************************************************************
+	 ** In DB we store approve-events for all addresses and ChainIDs. But we need to show to user
+	 ** only those that match their address and chainID
+	 *********************************************************************************************/
+	const chainFilteredAllowances = useMemo(() => {
+		const _formatedAllowances: TExpandedAllowance[] = [];
+		for (const allowance of cachedApproveEvents) {
+			_formatedAllowances.push({
+				...allowance,
+				args: {
+					owner: allowance.owner,
+					sender: allowance.sender,
+					value: allowance.value
+				}
+			});
+		}
+
+		const filteredAllowances = _formatedAllowances.filter(
+			item => item.args.owner === address && item.chainID === safeChainID
+		);
+
+		return filteredAllowances;
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [address, `${cachedApproveEvents}`, safeChainID]);
+
+	/**********************************************************************************************
+	 ** We sequentially apply filters to the allowances based on the provided filter object. First,
+	 ** we check for the presence of the 'unlimited' filter and apply it. Then, we move on to the
+	 ** 'asset' filter, ensuring the array is not empty before filtering by assets. The same
+	 ** process applies to the 'spender' filter.
+	 *********************************************************************************************/
+	const filteredAllowances = useDeepCompareMemo(() => {
+		const filters = configuration.allowancesFilters;
+
+		return chainFilteredAllowances?.filter(item => {
+			if (filters.unlimited.filter === 'unlimited') {
```

Comment:
Pretty sure that it would be easier to filter if the structure was like:
```
const filters: {
		limited: boolean | null;
		...
	};
```
so:
1. limited = true => filter out all the unlimited
2. limited = false => filter out all the limited
3. limited = null => show all

and this is applicable to all the filters here I guess

this filter.unlimited = 'limited' thing feels a bit off